packages/cw1155/src/event.rs

```diff
@@ -0,0 +1,42 @@
+use cosmwasm_std::{attr, Attribute, HumanAddr, Uint128};
+use cw0::Event;
+
+use crate::msg::TokenId;
+
+pub struct TransferEvent<'a> {
+    pub from: Option<&'a HumanAddr>,
+    pub to: Option<&'a HumanAddr>,
+    pub token_id: TokenId,
+    pub amount: Uint128,
+}
+
+impl Event for TransferEvent {
```

Comment:
Very nice idea. I like it.

So, the standard defines some Event types and the contracts can just instantiate these types and "serialize" them into the response. After this PR is merged, I will look to see how to adapt other contracts to this pattern.